packages/frontend/src/components/markdown-editor/markdown-editor.tsx

```diff
@@ -65,19 +67,51 @@ const emojiCompleter = {
 interface Props {
   contents: string;
   onContentsChange: (updatedValue: string) => any;
+  uploadFile?: (blob: File, name: string) => any;
 }
 
-export const MarkdownEditor = ({ contents, onContentsChange }: Props) => {
+export const MarkdownEditor = ({ contents, onContentsChange, uploadFile }: Props) => {
   const aceTheme = useColorModeValue('chrome', 'nord_dark');
 
   const [internalValue, setInternalValue] = useState(contents);
   const previousValueRef = useRef(contents);
 
+  const aceEditorRef: any = useRef();
+
   // Overwrite internal state if external contents change
   useEffect(() => {
     setInternalValue(contents);
   }, [contents]);
 
+  // Adding listener for the aceEditor on 'paste' event
+  useEffect(() => {
+    async function handlePasteEvent(arg: any) {
+      const items = (arg.clipboardData || arg.originalEvent.clipboardData).items;
+      if (items[0].type.startsWith('image/')) {
+        const blob = items[0].getAsFile();
+        if (uploadFile) {
+          const name = blob.name.startsWith('image.')
+            ? blob.name.replace('image', `pasted-${nanoid().substring(0, 6)}`)
+            : blob.name.replaceAll(' ', '-');
+          const cursorPos = aceEditorRef.current.editor.getCursorPosition();
+          aceEditorRef.current.editor.session.insert(cursorPos, `![${name}](${name})`);
+
+          const data = await uploadFile(blob, name);
+          if (data) {
+            console.log(`Image pasted with name: '${data.uploadSingleFile.name}'`);
+          }
+        }
+      }
+    }
+
+    if (aceEditorRef && aceEditorRef.current) {
+      document.querySelector(`#aceEditor`)?.addEventListener('paste', handlePasteEvent, true);
+      return function cleanup() {
+        document.querySelector(`#aceEditor`)?.removeEventListener('paste', handlePasteEvent, true);
+      };
```

Comment:
While investigating the issue on why the event listener was being triggered multiple times I found this post https://www.pluralsight.com/guides/how-to-cleanup-event-listeners-react that mentions that event listeners need to be removed after used in React to avoid precisely that issue.